modules/fbs/includes/fbs.reservation.inc

```diff
@@ -377,11 +385,26 @@ function fbs_reservation_options_submit($account, $values) {
 }
 
 /**
- * Implements provider reservation, default_interst_period.
+ * Implements hook_default_interest_period().
  *
- * @return int
- *   Default interest period for user.
+ * Tries to find the users default interest period based on the users profile2
+ * provider profile.
  */
-function fbs_reservation_default_interest_period() {
-  return variable_get('fbs_default_interest_period', 180);
+function fbs_reservation_default_interest_period($profile = NULL) {
```

Comment:
Then this function could be changed to:

```
function fbs_reservation_default_interest_period($account = NULL) {
  $account = is_null($account) ? $GLOBALS['user'] : $account;
  $profile = ding_user_provider_profile($user);

  // Get interest period from the profile.
  $wrapper = entity_metadata_wrapper('profile2', $profile);
  $value = $wrapper->field_fbs_interest_period->value();

  // If no period is selected, try getting default value.
  if (is_null($value) || empty($value)) {
    $value = variable_get('fbs_default_interest_period', 180);
  }
}
```
Or one could use `user_uid_optional_load` if the whole user object was required and not `$GLOBALS['user']`